http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/KubernetesServiceInterp.scala

```diff
@@ -235,9 +243,43 @@ class LeoKubernetesServiceInterp[F[_]: Parallel](
       _ <- publisherQueue.enqueue1(deleteMessage)
     } yield ()
 
-  private[service] def getSavableCluster(userInfo: UserInfo,
-                                         googleProject: GoogleProject,
-                                         now: Instant): Either[Throwable, SaveKubernetesCluster] = {
+  override def batchNodepoolCreate(userInfo: UserInfo, googleProject: GoogleProject, req: BatchNodepoolCreateRequest)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      hasPermission <- authProvider.hasPermission(ProjectSamResourceId(googleProject),
+                                                  ProjectAction.CreateApp,
+                                                  userInfo)
+      _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(userInfo.userEmail))
+
+      // create default nodepool with size dependant on number of nodes requested
+      saveCluster <- F.fromEither(getSavableCluster(userInfo, googleProject, ctx.now, Some(req.numNodepools)))
+      saveClusterResult <- KubernetesServiceDbQueries.saveOrGetForApp(saveCluster).transaction
+
+      // check if the cluster exists (we reject this request if it does)
+      createCluster <- saveClusterResult match {
+        case _: ClusterExists         => F.raiseError[CreateCluster](ClusterExistsException(googleProject))
+        case res: ClusterDoesNotExist => F.pure(CreateCluster(res.minimalCluster.id, res.defaultNodepool.id))
+      }
+      // create list of Precreating nodepools
+      eitherNodepoolsOrError = List.tabulate(req.numNodepools.value) { _ =>
+        getUserNodepool(createCluster.clusterId, userInfo, req.kubernetesRuntimeConfig, ctx.now)
+      }
+      nodepools <- eitherNodepoolsOrError.traverse(n => F.fromEither(n))
+      _ <- nodepoolQuery.saveAllForCluster(nodepools).transaction
+      dbNodepools <- KubernetesServiceDbQueries.getAllNodepoolsForCluster(createCluster.clusterId).transaction
+      allNodepoolIds = dbNodepools.map(_.id)
+      msg = BatchNodepoolCreateMessage(createCluster.clusterId, allNodepoolIds, googleProject, Some(ctx.traceId))
+      _ <- publisherQueue.enqueue1(msg)
```

Comment:
we're enqueuing pub sub messages in the  `KubernetesServiceInterp` instead of front leo?

Reply:
The `KubernetesServiceInterp` is front leo. You may be thinking of the GKEInterpreter which we worked on recently which is back leo